### DISCUSSION TITLE
Using Lossless cut with Gopro

### DISCUSSION THREAD

**Discussion by arnean** [mifi/lossless-cut#1570] [Show and tell]
I've found a nice way how to use Lossless cut with Gopro footage. 

My requirements for video editing are very basic - no effects, transitions or titles. I just need cut and merge. So Lossless cut should be perfect for this. Unfortunately it does not support HEVC videos that Gopro is creating. It is too slow with the realtime conversion plus there is no sound. But then I discovered that Gopro creates a low resolution clips for mobile preview. Those have LRV extension and they are very small H.264 encoded MP4 files - perfect for Lossless cut!

My workflow is following:

1. Insert MicroSD card into computer.
2. Launch Lossless cut 
3. Drag all LRV files into Lossless cut
4. Arrange them into correct order (Gopro has funky file naming standard)
5. Set working directory to my HDD
6. Batch merge the files from MicroSD to HDD
7. Repeat the same with full-size HEVC MP4 files. Now you have two merged files on your HDD - low resolution H.264 encoded and high-resolution HEVC version.
8. Open the low-res file and cut it. Since it is a low-res H.264 it is super fast to work on.
9. Once you are done, merge it and check it in video player.
10. Now copy the project file that was created for low-res version and rename it like to was created for high-res clip.
11. Open the project file with text editor - there is a clip name inside it. Change it to the name of the high-res clip. Close the project file.
12. Drag the high-res file into Lossless cut. The project file with all the cuts is opened.
13. Do the merge operation and close the Lossless cut
14. Now you have high-res version with all the cuts.

I like this workflow because it does not waste disk space at all. Also you get best quality because there is no re-encoding. It is super fast to do the "encoding". It is super fast to do the cutting. Lossless cut has absolutely fabulous keyboard support. I can work so quickly and precisely with it. Thank you for creating it!

**Comment by mifi** [comment]
Cool! One potential optimization for your work flow is to rename your .flv files to for example `ORIGINAL-html5ified-slow.mp4`, the place these files in the same directory as the originals (full resolution), and then open the original in losslesscut. It will then use the `ORIGINAL-html5ified-slow.mp4` file to preview, but when cutting, it will use the original file.

Example:
- `GOPRO123.mp4`
- `GOPRO123.lrv`  -> rename to `GOPRO123-html5ified-slow.mp4`

Maybe LosslessCut could automatically detect LRV files and auto-load them, however I need to know how is the specification for LRV file naming (how to know which original file do they refer to?) - Will it always be the same name as the original but instead ending with .LRV?

Another question is why doesn't hevc work in the first place, because LosslessCut does support HEVC now (if you have a hardware encoder and you enabled HEVC under "Settings")

**Reply by arnean** [reply]
> Cool! One potential optimization for your work flow is to rename your .flv files to for example `ORIGINAL-html5ified-slow.mp4`, the place these files in the same directory as the originals (full resolution), and then open the original in losslesscut. It will then use the `ORIGINAL-html5ified-slow.mp4` file to preview, but when cutting, it will use the original file.
> 
> Example:
> 
> * `GOPRO123.mp4`
> * `GOPRO123.lrv`  -> rename to `GOPRO123-html5ified-slow.mp4`

Very nice. I will try it out.

> Maybe LosslessCut could automatically detect LRV files and auto-load them, however I need to know how is the specification for LRV file naming (how to know which original file do they refer to?) - Will it always be the same name as the original but instead ending with .LRV?

High-res clips have name GX011029.MP4 and low-res is GL011029.LRV. So different extension and X vs L in filename. Gopro uses two-part sequence numbers. 

GX011029.MP4 is 01 + 1029. 1029 is number of the clip. Each time you start new recording new number is used. This should be primary sort key. 01 is number of the segment. It will keep the files below 4G. First segment is 01, second 02, etc. So this must be secondary sort key.

> Another question is why doesn't hevc work in the first place, because LosslessCut does support HEVC now (if you have a hardware encoder and you enabled HEVC under "Settings")

Oops :) I tried it last year, it didn't support HEVC and I found this workaround. I have been too lazy to update it (because it has been working) but I will do it now.

**Comment by arnean** [comment]
Tried out the HEVC support with the latest version. It works very well. I would say that my hack is not needed for me anymore.

**Reply by mifi** [reply]
ok cool . ithink that we don't need to implement this then, because all gopros probably use h264 or hevc, which are supported now

**Comment by dmitrisky** [comment]
What about stabilization tracks? I was not able to cut video in LoslessCut and then stabilize them in Gyroflow.

**Comment by arnean** [comment]
I realize I should have used a more specific title. If I understand it correctly LoslessCut doesn't know how to cut the Gopro metadata tracks. That would probably be a completely separate feature.

**Reply by mifi** [reply]
yes i believe gopro metadata tracks are proprietary and cannot be understood (or cut) by ffmpeg
